### PR TITLE
Issue860 rename class

### LINKED ISSUE #860: Rename CDL.SetPoints.HotWaterTemperatureReset
Consider changing `CDL.SetPoints.HotWaterTemperatureReset` to `SupplyReturnTemperatureReset` and update the documentation as it can be used for heating and chilled water applications.

### PR DESCRIPTION
**Pull request by JayHuLBL**
This closes #860.
- [x] renamed class `Buildings.Controls.OBC.CDL.SetPoints.HotWaterTemperatureReset` to `Buildings.Controls.OBC.CDL.SetPoints.SupplyReturnTemperatureReset`
- [x] renamed corresponded files, including the dymola and openModelica scripts, image name, reference data file

@mwetter 
It will be ready for your review when the tests passed.

### CODE REVIEW

File: Buildings/Controls/OBC/CDL/SetPoints/SupplyReturnTemperatureReset.mo [hunk 1]
```diff
@@ -76,7 +76,13 @@ shift the heating curve.
 </html>", revisions="<html>
 <ul>
 <li>
-January 10, 2016, by Milica Grahovac:<br/>
+January 3, 2020, by Jianjun Hu:<br/>
+Changed name from HotWaterTemperatureReset to SupplyReturnTemperatureReset.
```
Comment (mwetter) [inline Buildings/Controls/OBC/CDL/SetPoints/SupplyReturnTemperatureReset.mo]: use <code>...</code>.

File: Buildings/package.mo [hunk 2]
```diff
@@ -331,7 +331,16 @@ its class name ends with the string <code>Beta</code>.
     <b style=\"color:blue\">non-backward compatible</b> way:
     </p>
     <table class=\"releaseTable\" summary=\"summary\" border=\"1\" cellspacing=0 cellpadding=2 style=\"border-collapse:collapse;\">
-     <tr><td colspan=\"2\"><b>Buildings.Controls.OBC.ASHRAE.G36_PR1</b>
+    <tr><td colspan=\"2\"><b>Buildings.Controls.OBC.CDL</b>
+        </td>
+    </tr>
+    <tr><td valign=\"top\">Buildings.Controls.OBC.CDL.SetPoints.SupplyReturnTemperatureReset
+        </td>
+        <td valign=\"top\">Changed name from HotWaterTemperatureReset to SupplyReturnTemperatureReset.<br/>
```
Comment (mwetter) [inline Buildings/package.mo]: Use <code>...</code>

**Comment by JayHuLBL** [comment]
The comments were addressed. I will make an issue in IBPSA library.

**Comment by mwetter** [comment]
This is ready to merge when all tests pass.

**Pull request merged**